modules/calib3d/src/calibinit.cpp

```diff
@@ -734,51 +596,35 @@ int cvFindChessboardCorners( const void* arr, CvSize pattern_size,
 
     if( found )
     {
-      if ( pattern_size.height % 2 == 0 && pattern_size.width % 2 == 0 )
-      {
-        int last_row = (pattern_size.height-1)*pattern_size.width;
-        double dy0 = out_corners[last_row].y - out_corners[0].y;
-        if( dy0 < 0 )
+        if ( pattern_size.height % 2 == 0 && pattern_size.width % 2 == 0 )
         {
-          int n = pattern_size.width*pattern_size.height;
-          for(int i = 0; i < n/2; i++ )
-          {
-            CvPoint2D32f temp;
-            CV_SWAP(out_corners[i], out_corners[n-i-1], temp);
-          }
+            int last_row = (pattern_size.height-1)*pattern_size.width;
+            double dy0 = out_corners[last_row].y - out_corners[0].y;
+            if( dy0 < 0 )
+            {
+                int n = pattern_size.width*pattern_size.height;
+                for(int i = 0; i < n/2; i++ )
+                {
+                    CvPoint2D32f temp;
+                    CV_SWAP(out_corners[i], out_corners[n-i-1], temp);
+                }
+            }
         }
-      }
-      cv::Ptr<CvMat> gray;
-      if( CV_MAT_CN(img->type) != 1 )
-      {
-        gray.reset(cvCreateMat(img->rows, img->cols, CV_8UC1));
-        cvCvtColor(img, gray, CV_BGR2GRAY);
-      }
-      else
-      {
-        gray.reset(cvCloneMat(img));
-      }
-      int wsize = 2;
-      cvFindCornerSubPix( gray, out_corners, pattern_size.width*pattern_size.height,
-                          cvSize(wsize, wsize), cvSize(-1,-1),
-                          cvTermCriteria(CV_TERMCRIT_EPS+CV_TERMCRIT_ITER, 15, 0.1));
+        int wsize = 2;
+        CvMat old_img(img);
+        cvFindCornerSubPix( &old_img, out_corners, pattern_size.width*pattern_size.height,
```

Comment:
Calling C function instead of `cv::cornerSubPix` to use same `CvPoint2D32f * out_corners` array (not sure whether InputOutputArray supports this kind of arrays).